pytensor/tensor/basic.py

```diff
@@ -2789,13 +2781,59 @@ def vertical_stack(*args):
     _args = []
     for arg in args:
         _arg = as_tensor_variable(arg)
-        if _arg.type.ndim != 2:
-            raise ValueError("All arguments must have two dimensions")
         _args.append(_arg)
 
     return concatenate(_args, axis=0)
 
 
+def horizontal_stack(*args):
+    warnings.warn(
+        "horizontal_stack was renamed to hstack and will be removed in a future release",
+        FutureWarning,
+    )
+    return hstack(*args)
```

Comment:
Since the function signatures shouldn't be the same, I would just leave `horizontal_stack` as-is with the depreciation warning rather than aliasing to `hstack`